test/common.py

```diff
@@ -2191,18 +2191,17 @@ def run_browser(self, html_file, expected=None, message=None, timeout=None, extr
   def make_reftest(self, expected, manually_trigger=False):
     # make sure the pngs used here have no color correction, using e.g.
     #   pngcrush -rem gAMA -rem cHRM -rem iCCP -rem sRGB infile outfile
-    basename = os.path.basename(expected)
-    if os.path.abspath(os.path.dirname(expected)) != self.get_dir():
-      shutil.copyfile(expected, self.in_dir(basename))
     reporting = read_file(test_file('browser_reporting.js'))
+    shutil.copyfile(expected, 'expected.png')
     create_file('reftest.js', '''
       function doReftest() {
         if (doReftest.done) return;
         doReftest.done = true;
         var img = new Image();
-        img.onload = function() {
-          assert(img.width == Module.canvas.width, 'Invalid width: ' + Module.canvas.width + ', should be ' + img.width);
-          assert(img.height == Module.canvas.height, 'Invalid height: ' + Module.canvas.height + ', should be ' + img.height);
+        img.src = 'expected.png';
+        img.onload = () => {
```

Comment:
Is it safe to set `src` before `onload`? I have a vague worry about a race condition here, which is why I've written it in the opposite order.

Reply:
OK I'll revert that.